### PR TITLE
Enable field accessors to reference post aggregations in addition to …

### PR DESCRIPTION
**Pull request by michael-mclawhorn**
…aggregations.

The proximate need for this is that having metrics do not evaluate correctly on sketch metrics.  A group is going to try building a rewrite that adds synthetic sketch estimates and rewrites their havings to work around this bug.

It's not clear whether it's a better idea to wait for the sketch estimates to be fixed or to lift up this solution into Fili.  But they're going to take the hit of a first pass solution and maybe they'll contribute it back if it works well.

### CODE REVIEW

File: fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/postaggregation/FieldAccessorPostAggregation.java [hunk 1]
```diff
@@ -24,14 +24,14 @@
...
-    private final Aggregation aggregation;
+    private final MetricField metricField;
 
     /**
      * Constructor.
      *
      * @param aggregation  Aggregation to access
      */
-    public FieldAccessorPostAggregation(@NotNull Aggregation aggregation) {
+    public FieldAccessorPostAggregation(@NotNull MetricField aggregation) {
```
Comment (QubitPi) [inline fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/postaggregation/FieldAccessorPostAggregation.java]: Shall we change parameter name to `metricField`?

File: fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/postaggregation/FieldAccessorPostAggregation.java [hunk 2]
```diff
@@ -41,21 +41,40 @@ public FieldAccessorPostAggregation(@NotNull Aggregation aggregation) {
             throw new IllegalArgumentException(message);
         }
 
-        this.aggregation = aggregation;
+        this.metricField = aggregation;
```
Comment (QubitPi) [inline fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/postaggregation/FieldAccessorPostAggregation.java]: Shall we change `aggregation` to `metricField`?

File: fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/postaggregation/FieldAccessorPostAggregation.java (hunk 1 above)
Comment (asifmansoora) [reply fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/postaggregation/FieldAccessorPostAggregation.java]: @QubitPi its actually an `aggregation` that will be used here- hence the name

File: fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/postaggregation/FieldAccessorPostAggregation.java (hunk 2 above)
Comment (asifmansoora) [reply fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/postaggregation/FieldAccessorPostAggregation.java]: same applies here

**Pull request merged**